bulk_grades/api.py

```diff
@@ -359,37 +359,45 @@ def preprocess_row(self, row):
         operation = {}
         if row['user_id'] in self._users_seen:
             return operation
+
+        operation['new_override_grades'] = []
+        operation['course_id'] = self.course_id
+        operation['user_id'] = row['user_id']
+
         for key in row:
             if key.startswith('new_override-'):
                 value = row[key].strip()
                 if value:
                     short_id = key.split('-', 1)[1]
                     subsection = self._subsections[short_id][0]
-                    operation['user_id'] = row['user_id']
-                    operation['course_id'] = self.course_id
-                    operation['block_id'] = str(subsection.location)
+                    block_id = str(subsection.location)
                     try:
-                        operation['new_override'] = float(value)
+                        new_grade = float(value)
                     except ValueError as error:
                         raise ValidationError(_('Grade must be a number')) from error
-                    if operation['new_override'] < 0:
-                        raise ValidationError(_('Grade must be positive'))
+                    else:
+                        if new_grade < 0:
+                            raise ValidationError(_('Grade must be positive'))
+                        operation['new_override_grades'].append((block_id, new_grade))
```

Comment:
Using tuple to store the grades. Just seem to be convenient. I am open for suggest for better alternative if there is any. Especially we want to override multiple grade in a single request and modify `edx-platform/lms/djangoapps/grades/api.py::override_subsection_grade`

Reply:
I think this is a completely reasonable way to do it 👍